src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php

```diff
@@ -729,4 +807,254 @@ public static function provideReplaceIgnoreCase()
+    {
+        $instance = static::createFromString($origin)->ensureStart($prefix);
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideEnsureStart()
+    {
+        return [
+            ['', '', ''],
+            ['foo', 'foo', ''],
+            ['foo', '', 'foo'],
+            ['foo', 'foo', 'foo'],
+            ['foobar', 'foobar', 'foo'],
+            ['foobar', 'bar', 'foo'],
+            ['foo', 'foofoofoo', 'foo'],
+            ['foobar', 'foofoofoobar', 'foo'],
+            ['fooFoobar', 'Foobar', 'foo'],
+            ["foo\nfoo", "\nfoo", 'foo'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEnsureStartIgnoreCase
+     */
+    public function testEnsureStartIgnoreCase(string $expectedString, string $origin, $prefix)
+    {
+        $instance = static::createFromString($origin)->ignoreCase()->ensureStart($prefix);
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideEnsureStartIgnoreCase()
+    {
+        return [
+            ['', '', ''],
+            ['foo', 'foo', ''],
+            ['foo', '', 'foo'],
+            ['Foo', 'Foo', 'foo'],
+            ['Foobar', 'Foobar', 'foo'],
+            ['foobar', 'bar', 'foo'],
+            ['Foo', 'fOofoOFoo', 'foo'],
+            ['Foobar', 'fOofoOFoobar', 'foo'],
+            ["foo\nfoo", "\nfoo", 'foo'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEnsureEnd
+     */
+    public function testEnsureEnd(string $expectedString, string $origin, $suffix)
+    {
+        $instance = static::createFromString($origin)->ensureEnd($suffix);
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideEnsureEnd()
+    {
+        return [
+            ['', '', ''],
+            ['foo', 'foo', ''],
+            ['foo', '', 'foo'],
+            ['foo', 'foo', 'foo'],
+            ['foobar', 'foobar', 'bar'],
+            ['foobar', 'foo', 'bar'],
+            ['foo', 'foofoofoo', 'foo'],
+            ['foobar', 'foobarbarbar', 'bar'],
+            ['fooBarbar', 'fooBar', 'bar'],
+            ["foo\nfoo", "foo\n", 'foo'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideEnsureEndIgnoreCase
+     */
+    public function testEnsureEndIgnoreCase(string $expectedString, string $origin, $suffix)
+    {
+        $instance = static::createFromString($origin)->ignoreCase()->ensureEnd($suffix);
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideEnsureEndIgnoreCase()
+    {
+        return [
+            ['', '', ''],
+            ['foo', 'foo', ''],
+            ['foo', '', 'foo'],
+            ['foo', 'foo', 'foo'],
+            ['fooBar', 'fooBar', 'bar'],
+            ['foobar', 'foo', 'bar'],
+            ['fOo', 'fOofoOFoo', 'foo'],
```

Comment:
this is failing, current result is `foo`. this is inconsistent with `ignoreCase()->ensureStart()` where the case is preserved.

Reply:
thanks, should be good now